hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/BasePreCommit.java

```diff
@@ -552,6 +619,30 @@ void add(EntityAndPendingNotifications<BE, ?> entity) {
             found.notifications.addAll(entity.getNotifications());
         }
 
+        ProcessingTree<BE> getChild(RelativePath childPath) {
+            Iterator<Path.Segment> segments = childPath.getPath().iterator();
+            Set<ProcessingTree<BE>> children = this.children;
+
+            CanonicalPath.Extender childCp = cp.modified();
+            ProcessingTree<BE> current = null;
+            while (segments.hasNext()) {
+                Path.Segment seg = segments.next();
+                CanonicalPath expectedCp = childCp.extend(seg.getElementType(), seg.getElementId()).get();
+                childCp = expectedCp.modified();
+
+                current = null;
+                for (ProcessingTree<BE> c : children) {
+                    if (expectedCp.equals(c.cp)) {
+                        current = c;
+                        children = c.children;
+                        break;
+                    }
+                }
+            }
+
+            return segments.hasNext() ? null : current;
```

Comment:
Maybe I'm too tired, but... segments.hasNext() will always be false, no?
I'm not sure to understand the algorithm globally... shouldn't we have a breaking condition on the "while" loop?